### PR TITLE
Ignore kmod fact read errors

### LINKED ISSUE #79: Warning after upgrading to v3.1.0
After upgrading to v3.1.0, puppet started issuing warning each run

```
Warning: Facter: Operation not permitted @ io_fread - /sys/module/kdb_main/parameters/enable_nmi
Warning: Facter: Permission denied @ rb_sysopen - /sys/module/md_mod/parameters/new_array
Warning: Facter: Permission denied @ rb_sysopen - /sys/module/ipmi_si/parameters/hotmod

```

Puppet 6.25.1
Checked on CentOS 7, RHEL 8 and Fedora 35

### PR DESCRIPTION
**Pull request by jcpunk**
#### Pull Request (PR) description
The more targeted `rescue` seems to have a few edges I didn't expect.  This PR just puts all the errors into the facter debug level.  This should let folks track them if needed, but hide the results for the various edges folks are encountering.

#### This Pull Request (PR) fixes the following issues
Fixes #79

### CODE REVIEW

**Comment by silug** [comment]
I can confirm that this eliminates the
```
Warning: Facter: Operation not permitted @ io_fread - /sys/module/kdb_main/parameters/enable_nmi
```
message that I see on CentOS 7 without the change.

File: lib/facter/kmod.rb [hunk 1]
```diff
@@ -24,10 +24,8 @@
 
               kmod[directory]['parameters'][param] = File.read("/sys/module/#{directory}/parameters/#{param}").chomp
```
Comment (smortex) [inline lib/facter/kmod.rb:25]: Should we not catch exceptions around this single line?

I feel like if` /sys/module/#{directory}/parameters` contains multiple files, and reading one fail, we stop processing the directory and the fact will only have a partial view of the parameters.  Also I am not sure the subset of parameters we get will be the same all the time.

Comment (jcpunk) [reply lib/facter/kmod.rb:25]: Good call, I've just reworked the rescue bits, do they look better now?

Comment (smortex) [reply lib/facter/kmod.rb:25]: LGTM!

File: lib/facter/kmod.rb [hunk 2]
```diff
@@ -16,30 +16,28 @@
...
         if File.directory?("/sys/module/#{directory}/holders")
-          begin
-            Dir.foreach("/sys/module/#{directory}/holders") do |used|
-              next if ['.', '..'].include?(used)
+          Dir.foreach("/sys/module/#{directory}/holders") do |used|
+            next if ['.', '..'].include?(used)
 
+            begin
               kmod[directory]['used_by'] << used
+            rescue StandardError => e
```
Comment (smortex) [inline lib/facter/kmod.rb:38]: This begin / rescue protect code that add a String in an Array… It should never fail :smile: 

I don't think listing holders could be a problem so maybe we can completely get rid of this one?

**Review by smortex — COMMENTED** [review]
Better I think, spotted another minor thing you may want to review.

File: lib/facter/kmod.rb (hunk 2 above)
Comment (jcpunk) [reply lib/facter/kmod.rb]: :) took a stab at solving and pushed up.

**Review by smortex — APPROVED** [review]
LGTM, Thanks!

**Comment by jcpunk** [comment]
I'll look into reworking this as something less intrusive, but for now can we get this merged so folks at least aren't actively bugged with the read errors?

**Review by bastelfreak — APPROVED** [review]
looks good to me. Some people around that want to test it on their workstation?

**Comment by gcoxmoz** [comment]
Runs cleanly on both CentOS7 and 8, does what it says, comments make sense explaining why values/errors are being suppressed.

**Pull request merged**